samples/rust/src/main.rs

```diff
@@ -3,18 +3,62 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 use std::env;
+use std::fs::File;
+use std::io::{Read, Write};
 use std::path::PathBuf;
 use std::thread;
 use std::time::Duration;
 
 use tempfile::Builder;
 
-use glean::{ClientInfoMetrics, Configuration};
+use flate2::read::GzDecoder;
+use glean::{net, ClientInfoMetrics, ConfigurationBuilder};
 
 pub mod glean_metrics {
     include!(concat!(env!("OUT_DIR"), "/glean_metrics.rs"));
 }
 
+#[derive(Debug)]
+struct MovingUploader(String);
+
+impl net::PingUploader for MovingUploader {
```

Comment:
This appears to be some extra stuff not strictly needed for the object type, no?

Reply:
Ah, I see, this is in a different commit that's kindly explaining what it's for. Gorram Github review UI being bad at git.

Reply:
Ah yes, I added that because I had it lying around for testing anyway.